src/Native/Unix/System.Native/pal_interfaceaddresses.c

```diff
@@ -80,16 +122,11 @@ int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onIpv4Found,
                 struct sockaddr_in* sain = (struct sockaddr_in*)current->ifa_addr;
                 memcpy_s(iai.AddressBytes, sizeof_member(IpAddressInfo, AddressBytes), &sain->sin_addr.s_addr, sizeof(sain->sin_addr.s_addr));
 
-                // Net Mask
-                IpAddressInfo maskInfo;
-                memset(&maskInfo, 0, sizeof(IpAddressInfo));
-                maskInfo.InterfaceIndex = interfaceIndex;
-                maskInfo.NumAddressBytes = NUM_BYTES_IN_IPV4_ADDRESS;
-
                 struct sockaddr_in* mask_sain = (struct sockaddr_in*)current->ifa_netmask;
-                memcpy_s(maskInfo.AddressBytes, sizeof_member(IpAddressInfo, AddressBytes), &mask_sain->sin_addr.s_addr, sizeof(mask_sain->sin_addr.s_addr));
+                // ifa_netmask can be NULL according to documentation, probably P2P interfaces.
+                iai.PrefixLength = mask_sain != NULL ? mask2prefix((uint8_t*)&mask_sain->sin_addr.s_addr, NUM_BYTES_IN_IPV4_ADDRESS) : NUM_BYTES_IN_IPV4_ADDRESS * 8;
```

Comment:
So PrefixLength uses bits as a unit?

Reply:
correct. so for example 10.11.12.13 with netmask 255.255.255.0 becomes represented as 10.11.12.13/24. For IPv6 the whole netmask was dropped as impractical and only slash notation is used. New Linux tools moved towards that as well:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: enp0s5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 00:1c:42:20:46:36 brd ff:ff:ff:ff:ff:ff
    inet 10.211.55.17/24 brd 10.211.55.255 scope global enp0s5
       valid_lft forever preferred_lft forever
    inet6 2002:2c26:f4e4:0:21c:42ff:fe20:4636/64 scope global mngtmpaddr dynamic
       valid_lft 2591505sec preferred_lft 604305sec
    inet6 fe80::21c:42ff:fe20:4636/64 scope link
       valid_lft forever preferred_lft forever
```